### PR TITLE
Make Travis read the CXXFLAGS enviroment variable

### PR DESCRIPTION
**Pull request by arrufat**
If this worked, the code should not compile with C++17. See #2365.

### CODE REVIEW

**Comment by pfeatherstone** [comment]
@arrufat Since the checks passed, does this mean the fix didn't work?

**Comment by arrufat** [comment]
Yes :(

File: dlib/travis/build-and-test.sh [hunk 1]
```diff
@@ -7,11 +7,16 @@ if [ -n "${MATRIX_EVAL+set}" ]; then
     eval "${MATRIX_EVAL}"
 fi
 
+CXX_FLAGS="-std=c++11"
+if [ ! -z ${CXXFLAGS+x} ]; then
```
Comment (davisking) [inline dlib/travis/build-and-test.sh:11]: Why is there that +x there?  
```suggestion
if [ ! -z ${CXXFLAGS} ]; then
```
Why not do this?

Comment (davisking) [reply dlib/travis/build-and-test.sh:11]: Heh, I don't recall why I did that odd `+set` thing on the one above either :|

Comment (davisking) [inline dlib/travis/build-and-test.sh:11]:
```suggestion
if [ ! -z ${CXXFLAGS+set} ]; then
```
Do this though, so it's the same style as the one above.

Comment (arrufat) [reply dlib/travis/build-and-test.sh]: Yes, that was my first try...

**Pull request merged**